File: frontend/src/api/jobs.ts
import { apiFetch } from "./client"

export interface BulkJobItem {
  id: string
  account_id: string
  status: string
  attempt_count: number
  error_message: string | null
  flood_wait_seconds: number | null
  started_at: string | null
  finished_at: string | null
}

export interface BulkJob {
  id: string
  job_type: string
  status: string
  total_items: number
  succeeded_count: number
  failed_count: number
  created_at: string
  started_at: string | null
  finished_at: string | null
}

export interface BulkJobDetail extends BulkJob {
  items: BulkJobItem[]
}

export function listJobs() {
  return apiFetch<BulkJob[]>("/api/bulk-jobs")
}

export function getJob(jobId: string) {
  return apiFetch<BulkJobDetail>(`/api/bulk-jobs/${jobId}`)
}

export interface CreateJobPayload {
  account_ids: string[]
  first_name?: string
  last_name?: string
  bio?: string
  username?: string
  per_account?: Record<string, Record<string, string>>
}

export function createJob(payload: CreateJobPayload, photo?: File | null) {
  const form = new FormData()
  form.set("payload", JSON.stringify(payload))
  if (photo) form.set("photo", photo)
  return apiFetch<BulkJob>("/api/bulk-jobs", { method: "POST", body: form })
}

File: backend/app/schemas/jobs.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class BulkJobCreate(BaseModel):
    account_ids: list[str]
    first_name: str | None = None
    last_name: str | None = None
    bio: str | None = None
    username: str | None = None
    per_account: dict[str, dict] | None = None  # account_id -> field overrides


class BulkJobItemOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    account_id: str
    status: str
    attempt_count: int
    error_message: str | None
    flood_wait_seconds: int | None
    started_at: datetime | None
    finished_at: datetime | None


class BulkJobOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    job_type: str
    status: str
    total_items: int
    succeeded_count: int
    failed_count: int
    created_at: datetime
    started_at: datetime | None
    finished_at: datetime | None


class BulkJobDetailOut(BulkJobOut):
    items: list[BulkJobItemOut]
